### ISSUE TITLE
Inconsistent deserialization for OneOrMany properties on derived classes

### ISSUE THREAD

**Issue by TadDockery** [RehanSaeed/Schema.NET#110]
### Describe the bug

I have some properties of `OneOrMany<T>` that deserialize when they're single valued, but not when array valued; but others that deserialize fine in either situation.

### To Reproduce

I'm working with three derived classes: the first extends PostalAddress with properties of type `OneOrMany<IAdministrativeArea>`, and is called `ExtendedAddress`; the second extends Thing with some basic tracking, and is called `ExtendedThing`; the third contains properties of both those types, and is called `Project`. Here's the relevant parts of each class:

```
public partial class ExtendedThing : Thing, IExtendedThing
{
		[DataMember(Name = "@type", Order = 1)]
		public override string Type => "ExtendedThing";
  // The tracking properties aren't relevant right now
}

public partial class ExtendedAddress : PostalAddress, IExtendedAddress
{
		[DataMember(Name = "@type", Order = 1)]
		public override string Type => "ExtendedAddress";
		[DataMember(Name = "addressCounty", Order = 506)]
		[JsonConverter(typeof(ValuesJsonConverter))]
		public OneOrMany<IAdministrativeArea> AddressCounty { set; get; }
}

public partial class Project : ExtendedThing, IProject
{
		[DataMember(Name = "@type", Order = 1)]
		public override string Type => "Project";
		[DataMember(Name = "affectedSite", Order = 206)]
		[JsonConverter(typeof(ValuesJsonConverter))]
		public OneOrMany<ExtendedThing> AffectedSite { set; get; }
		[DataMember(Name = "projectAddress", Order = 210)]
		[JsonConverter(typeof(ValuesJsonConverter))]
		public OneOrMany<ExtendedAddress> ProjectAddress { set; get; }
}
```

I want these Projects to be assigned to a ReceiveAction as the Object property; but since that doesn't deserialize except as a Thing, I've been serializing and assigning the string to the ReceiveActions' Object's AlternateName. After transport I do this: `inputObject.Object = JsonConvert.DeserializeObject<Project>(inputObject.Object.First().AlternateName.First().ToString());`. That worked originally, but doing some regression testing I now see that a single AffectedSite deserializes as an ExtendedThing, and an AffectedSite array deserializes as null. The AddressCounty, meanwhile, deserializes properly whether it's a single value or an array. In other words, the following string

```
"{\"@context\":\"http://some.site.io\",\"@type\":\"Project\",\"affectedSite\":[{\"@type\":\"ExtendedThing\",\"@id\":\"http://some.site.io/Site/1\",\"name\":\"Example Site 1\"},{\"@type\":\"ExtendedThing\",\"@id\":\"http://some.site.io/Site/2\",\"name\":\"Example Site 2\"}],\"projectAddress\":{\"@type\":\"ExtendedAddress\",\"addressCounty\":[{\"@type\":\"AdministrativeArea\",\"name\":\"County 1\",\"alternateName\":\"1\"},{\"@type\":\"AdministrativeArea\",\"name\":\"County 2\",\"alternateName\":\"2\"}]}}"
```

should deserialize like

```
Project {
  AffectedSite [Site 1, Site 2],
  ProjectAddress { AddressCounty [County 1, County 2] }
}
```

but instead deserializes like

```
Project {
  ProjectAddress { AddressCounty [County 1, County 2] }
}
```

**Comment by Turnerj** [comment]
I encountered something similar with a difference between how array values are handled vs single values. I can't say for certain the cause is the same here however I am in the process of rewriting the `ValuesJsonConverter` to fix a number of issues (see #109).

If you want to try out to see if that solves the issue for you, try downloading a build from here: https://dev.azure.com/schema-net/Schema.NET/_build/results?buildId=363&view=artifacts&type=publishedArtifacts (its the current latest from that PR).

**Comment by TadDockery** [comment]
That's definitely it, thanks!
